rn/Teacher/ios/TeacherTests/Attendance/AttendanceViewControllerTests.swift

```diff
@@ -0,0 +1,148 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import XCTest
+@testable import Core
+@testable import Teacher
+import TestsFoundation
+
+class AttendanceViewControllerTests: TeacherTestCase {
+    let context = ContextModel(.course, id: "1")
+    var controller: AttendanceViewController!
+    var navigation: UINavigationController!
+
+    override func setUp() {
+        super.setUp()
+
+        let now = DateComponents(calendar: .current, timeZone: .current, year: 2019, month: 10, day: 31).date!
+        Clock.mockNow(now)
+
+        controller = AttendanceViewController(context: context, toolID: "1")
+        controller.session.state = .active(MockURLSession())
+
+        navigation = UINavigationController(rootViewController: controller)
+
+        api.mock(GetCustomColorsRequest(), value: APICustomColors(custom_colors: [
+            context.canvasContextID: "#008EE2", // electric
+        ]))
+        api.mock(GetCourseRequest(courseID: context.id), value: .make())
+        api.mock(GetCourseSectionsRequest(courseID: context.id, perPage: 100), value: [
+            .make(),
+            .make(name: "section 2"),
+        ])
```

Comment:
For future reference, you can pass in a `Store` to `api.mock` so you don't have to match the request exactly. Eg:

```swift
api.mock(controller.colors, value: ...)
api.mock(controller.course, value: .make())
api.mock(controller.sections, value: [...])
```

Reply:
That's nice. Looks like I need to fix a swiftlint error, so I may as well change these too.